pkg/frontend/transport/handler.go

```diff
@@ -295,3 +305,33 @@ func statsValue(name string, d time.Duration) string {
 	durationInMs := strconv.FormatFloat(float64(d)/float64(time.Millisecond), 'f', -1, 64)
 	return name + ";dur=" + durationInMs
 }
+
+// ActivityTracker is the minimum abstraction for *activitytracker.ActivityTracker, allowing us to replace it in tests.
+type ActivityTracker interface {
+	Insert(func() string) (activityIndex int)
+	Delete(index int)
+}
+
+func httpRequestActivityGenerator(request *http.Request) func() string {
+	return func() string {
+		tenantIDs, err := tenant.TenantIDs(request.Context())
+		tenantID := tenant.JoinTenantIDs(tenantIDs)
+		if err != nil {
+			tenantID = fmt.Sprintf("unknonwn, err: %v", err)
```

Comment:
```suggestion
			tenantID = "unknown"
```

"activities" with invalid tenant or form parsing errors are not interesting -- they get rejected quickly, and we don't need to put errors into activity description.

Reply:
Agree on the fact that they're not interesting, will apply this.

Reply:
Changed in https://github.com/grafana/mimir/pull/3561/commits/8eb8d50dd36f0bbb87a00e3776049581e2f4dc70

Also removed the `fmt.Sprintf` call in favour of a cheaper `strings.Join`.